server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -417,9 +420,28 @@ protected ValuesSourceAggregatorFactory innerBuild(
         AggregatorFactory parent,
         AggregatorFactories.Builder subFactoriesBuilder
     ) throws IOException {
-        DateHistogramAggregationSupplier aggregatorSupplier = context.getValuesSourceRegistry().getAggregator(REGISTRY_KEY, config);
+        final String indexMode = context.getIndexSettings().getSettings().get(IndexSettings.MODE.getKey());
+        final DateIntervalWrapper.IntervalTypeEnum dateHistogramIntervalType = dateHistogramInterval.getIntervalType();
+        if (IndexMode.TIME_SERIES.getName().equals(indexMode)
+            && DateIntervalWrapper.IntervalTypeEnum.CALENDAR.equals(dateHistogramIntervalType)) {
+            throw new UnsupportedAggregationOnDownsampledField(
+                config.getDescription()
+                    + " is not supported for aggregation ["
+                    + this.getName()
+                    + "] with interval type ["
+                    + dateHistogramIntervalType.getPreferredName()
+                    + "]"
+            );
+        }
 
         final ZoneId tz = timeZone();
+        if (IndexMode.TIME_SERIES.getName().equals(indexMode) && ZoneId.of("UTC").equals(tz) == false) {
+            throw new UnsupportedAggregationOnDownsampledField(
```

Comment:
Should we use a different exception here?

Reply:
I'm not sure! I'd go with the same exception for now and see if the kibana folk want to change it.